modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml

```diff
@@ -1237,6 +1249,1067 @@ paths:
             summary: "Gets a list of channels."
             tags:
                 - Channel
+    "/channels/by-externalReferenceCode/{externalReferenceCode}/products/{productExternalReferenceCode}/pins":
```

Comment:
lets call this    `/channels/by-externalReferenceCode/{channelExternalReferenceCode}/products/by-externalReferenceCode/{productExternalReferenceCode}/pins`

so (for all the parameters put the by-externalReferenceCode and name the parameters accordling. I think I changed my idea compared with previous rules but it's better. Sorry for that

Reply:
Remember also to update the name of the variables the resourceImpl methods like they will be generated by the buildRest in the base files